client/src/components/application/Form.tsx

```diff
@@ -53,6 +56,7 @@ export default function Form({
   showErrorModal: boolean;
 }) {
   const dispatch = useDispatch();
+  const { chains } = useNetwork();
```

Comment:
```
  const { chains } = useNetwork();
  ...
  const chainInfo = chains.find((i) => i.id === props.chainID);
```

can be replaced by
```
const { chain } = useNetwork();
```
the `chain` object equals `chainInfo`

Reply:
The problem with doing it this way is it breaks the tests that have chain Id 5 hardcoded, the test fails due to it using the current network you are connected to. We could refactor this and the tests later.